qiita_db/util.py

```diff
@@ -1730,6 +1733,34 @@ def _get_filehandle(filepath_or, *args, **kwargs):
     if _is_string_or_bytes(filepath_or):
         if h5py.is_hdf5(filepath_or):
             fh, own_fh = h5py.File(filepath_or, *args, **kwargs), True
+        elif filepath_or.endswith('.xlsx'):
```

Comment:
This is going to fail when `filepath_or` is not a string because it won't have the `endswith` method, right? 

I think the XLSX parsing should be moved elsewhere this function seems too generic as it should only open a file path or file handle. Is there any motivation to have this here instead of in the sample/prep information parsing?

Reply:
Right! I initially was gonna add the code in `load_template_to_dataframe` but realized that it really just calls this function (see, https://github.com/biocore/qiita/blob/master/qiita_db/metadata_template/util.py#L100) as it just calls the open_file method. BTW note that the same method is available to open hdf5, text, or iostring. Thus, I believe is fine to fail if not a string as the other code should be able to handle and parse correctly those errors. Thoughts?

Reply:
Fair enough, the use of this function is rather limited (see these [search results](https://github.com/biocore/qiita/search?utf8=✓&q=open_file&type=)). It still seems too odd to me that a function that's intended to return a file handle will parse a file write it as a txt file and return that file handle.

Reply:
Well, the idea of parsing the xmlx is that it will be transformed to txt and then use the regular prep/sample code ... ¯\_(ツ)_/¯ ... I don't have a better place to move it but if you do, let me know.

Reply:
I think it makes more sense to have this outside of this function. I see as two very different things, (1) opening a file and (2) parsing a file. And this function opens the file and shouldn't be parsing it. Is there any technical limitation to putting this in load_template_to_dataframe? Perhaps as a helper function somewhere in there?